### ISSUE TITLE
Switch to BlazeHTML

### ISSUE THREAD

**Issue by edsko** [haskell/hackage-server#13]

**Comment by kanishka-azimi** [comment]
This seems done. Lucid next? Considering Lucid's dependencies, it would add a dependency on mmorph.

**Comment by gbaz** [comment]
i think we should close this as we've been pushing to move more towards templates, and i think the remaining fragment generation is not a major issue?

**Comment by 414owen** [comment]
Is the consensus still to move towards templates?
Personally, I'm opposed to all templating.
They're inevitable half-baked, with inconsistent features tacked on when someone decides they need them, they're untyped, they compose awfully, don't lend themselves towards abstraction, and the dev experience (highlighting/language server) is awful.
I enjoy writing haskell, I don't enjoy writing {jinja,gomplate,st}

**Comment by gbaz** [comment]
We moved towards st templates a fair amount already. A big advantage is we can reload them without forcing a reload of the server, which is helpful for stability and uptime. There's arguments a number of ways about the "right" way to do things, but imho of the major issues hackage faces, the choice of html generation is not among them. What we have now is basically fine, and switching would be a bunch of work with little payout. Closing the ticket on this basis.
